crates/ruma-identity-service-api/src/invitation/store_invitation/v2.rs

```diff
@@ -0,0 +1,95 @@
+//! [POST /_matrix/identity/v2/store-invite](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-store-invite)
+
+use ruma_api::ruma_api;
+use ruma_common::thirdparty::Medium;
+use ruma_identifiers::{MxcUri, RoomAliasId, RoomId, UserId};
+
+ruma_api! {
+    metadata: {
+        description: "Store pending invitations to a user's 3PID.",
+        method: POST,
+        name: "store_invitation",
+        path: "/_matrix/identity/v2/store-invite",
+        authentication: AccessToken,
+        rate_limited: false,
+    }
+
+    request: {
+        /// The literal string `email`.
+        pub medium: &'a Medium,
```

Comment:
I really wonder what to do about this field. The type makes sense, but also has `msisdn` as a possible value, which the docs say isn't supported for this endpoint. If we want to keep this type, we should probably make it a param to the constructor as well, but add a big warning on the field's docs that only `email` is supported. Alternatively, we could create a separate `Medium` enum with just one variant inside this module (or somewhere else in the crate if this same restriction applies to other identity service endpoints).

Reply:
Yeah, I don't know about this one either. I'd like to use the shared type so that if other media become supported, we won't have to make a breaking change. On the other hand, just having a comment doesn't prevent developers from using the wrong value at compile time...

For an implementing server, we sort of expect the developer to read the specification, which clearly states the limitation, so I don't think we'd need to worry about them. I guess a client developer should read it as well...

If I had to vote, I'd lean towards not creating a separate type.

Reply:
Alright, sounds good. I think our own docs should also be a bit clearer though. The tagline (first paragraph) should be a regular "medium for this …" thing, but we should have a second paragraph explaining that currently, email is the only medium that is supported according to the spec.

Also this means that my previous suggestions to leave the `medium` out of `fn new`s parameters is invalid. We should have it as a parameter. I guess we could also offer an `email` constructor that just delegates `Self::new(&Medium::Email, token, public_keys, display_name)` so users don't need to import `Medium`.

Reply:
Looking around, this is the only endpoint from IS API which is not deprecated and which has this restriction.

The only other endpoint I found to have this restriction was in the federation API: https://matrix.org/docs/spec/server_server/r0.1.4#put-matrix-federation-v1-3pid-onbind

There we just use `Medium` with a comment explaining that only `email` is supported:
```rust
    request: {
        /// The type of third party identifier. Currently only "email" is a possible value.
        pub medium: Medium,
```

Additionally the constructor omits the `medium` parameter. The same goes for the `ThirdPartyInvite` struct defined in the same module.

Should I remain consistent with that example?

Reply:
I've updated that endpoint, now you can do things the same way as there 😄

ecad815a5b1c0574d905e55a109eef2747987f93